src/redux/zipReducer.js

```diff
@@ -0,0 +1,50 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+// Thunk makes API call, returns action of the type specified by the first parameter
+// In the slice, extraReducers define different actions based on the 3 possible states of the returned promise and execute reducers for them
+export const zipFetch = createAsyncThunk(
+  'zipcodes/zipFetch',
+  //payload creator
+  async () => {
+    const url = "http://localhost:3001/zipcodes"
+    try {
+      const res = await axios.get(url);
+      return res.data; 
+    } catch(err) {
+        console.error("There was a problem fetching zipcode data" + err);
+        return {}
+    }  
+  }
+);
+
+// Then, handle actions in your reducers:
+const zipSlice = createSlice({
+  name: 'zipcodes',
+  initialState: { zipcodes: {}, status: 'idle', error: null },
+  reducers: {
```

Comment:
Same note as countyReducer.js - should the (lack of) need for actions in this reducer be commented on? Feel free to delete if you think it's fine as is.